pkg/client-sdk/client.go

```diff
@@ -229,6 +229,49 @@ func (a *arkClient) ListVtxos(
 	return
 }
 
+func (a *arkClient) GetTransactionHistory(ctx context.Context) ([]Transaction, error) {
```

Comment:
We can change the logic of this function as follows:
```
GetTransactionHistory {
   init txs map[string]Transaction // used to return the []Transaction at the end
   init help map[string]string // used to update a pending tx to claimed if needed
   
   loop spentVtxos {
      if vtxo is pending {
          help[vtxo.SpentBy] = vtxo.RedeemTx.Txid
          txs[vtxo.RedeemTx.Txid] = Transaction{...}
          continue
      }
      txs[vtxo.RoundTxid] = Transaction{...}
   }
   
   loop spendableVtxos {
       if vtxo is pending {
           txs[vtxo.RedeemTxid] = Transaction{...}
           continue
       }
       if help[vtxo.RoundTxid] {
           txs[help[vtxo.RoundTxid]].Pending = false
       } else {
           txs[vtxo.RoundTxid] = Transaction{...}
       }
}
```

Reply:
This way we prevent having multiple txs representing a single claim.
Not completely convinced of using just a Pending field to represent pending/claimed status of an async payment... maybe better to have in Transaction both Pending and Claimed bool field